### ISSUE TITLE
Support SQL Aliases

### ISSUE THREAD

**Issue by LarryC42** [dotnet/SqlClient#997]
### SQL Aliases don't work in .Net core 

Issue 14945 was closed with resolutions provided but not implemented.  Please implement or provide a program agnostic resolution that allows thousands of desktops to connect to a new refreshed instances of SQL Server without manual intervention.  .Net core is not suitable for SQL Server database applications in the enterprise as the client currently exists.  DNS alone is not adequate.  Oracle and Postgres both provide equivalent functionality using an external lookup file.  Whether it's in the file system or registry, aliases are an important management tool.

**Comment by JRahnama** [comment]
@roji can you point me to the right direction on Postgres's implementation of this feature?

**Comment by roji** [comment]
@JRahnama PostgreSQL doesn't have something that corresponds exactly to SQL Server aliases - definitely nothing that's stored in the Windows registry. The similar thing in PostgreSQL is called the [connection service file](https://www.postgresql.org/docs/current/libpq-pgservice.html), which is just text-file-based and so cross-platform.

It should be possible to support SQL Server Aliases in .NET Core when running on Windows by taking a dependency on [Microsoft.Win32.Registry](https://www.nuget.org/packages/Microsoft.Win32.Registry), checking at runtime whether you're on Windows or not (e.g. with [OperatingSystem.IsWindows](https://docs.microsoft.com/en-us/dotnet/api/system.operatingsystem.iswindows?view=net-5.0#System_OperatingSystem_IsWindows)), and checking the registry. The main issue with that would be the extra nuget dependency which is only relevant for this feature and only on Windows, which isn't great.

**Comment by ErikEJ** [comment]
An extension Nuget package with SQL Server Client Windows specific features?

**Comment by roji** [comment]
@ErikEJ sure, though that would require a new plugin/extension point etc.

**Comment by LarryC42** [comment]
Does anyone have access to the SQL Server team for feedback?  I've reached out the the SQL Server community as well here https://feedback.azure.com/forums/908035-sql-server/suggestions/43020111-sql-server-connection-aliases.  It feels like something that would benefit other languages as well as other platforms.  I think the stated problem is relatively simple:  Translate a name to a new name and port.  A simple text file with name,name,port would be sufficient, could be implemented without dependencies and would work cross platform and cross languages.  The trick then is how does the client locate that file?  More importantly though is getting the SQL Server community to embrace it as widely as possible so that the solution remains agnostic to applications, languages, and platforms.  Without the SQL Server team buying into the solution it feels like the only right thing is to implement the windows registry because all the official SQL Server drivers use that mechanism.

**Comment by David-Engel** [comment]
> Does anyone have access to the SQL Server team for feedback?

Yes. I've added this topic to discuss internally as a cross-driver, cross-platform idea.

> The trick then is how does the client locate that file?

A simple environment variable, most likely.

Just to set expectations, though, this would have to compete with other feature demands. And I don't see many votes for this here or on the SQL Server feedback forum. I do recognize the benefit of aliases, in general, and that they've been supported on Windows for a long time via the registry. There is obviously a need to bring the current Windows-only feature into the cross-platform world and this would remove a hurdle for those moving from Windows-only to cross-platform environments.

**Comment by JRahnama** [comment]
@roji thanks for the update.

**Comment by Pertsa2000** [comment]
With PowerShell 7.2 released, this problem hits there too -- and now makes a pain to upgrade anything to newer powershell because everything uses aliases, since it's so much easier to handle migration to different servers that way...

**Comment by dsartain18** [comment]
I've run into this issue as well on another ticket that was closed as it duplicated this one.
